src/bootstrap/bin/rustc.rs

```diff
@@ -127,11 +127,15 @@ fn main() {
             }
         }
 
+        // Cargo doesn't pass RUSTFLAGS to proc_macros:
+        // https://github.com/rust-lang/cargo/issues/4423
+        // Thus, if we are on stage 0, we explicitly set `--cfg=bootstrap`.
+        // If we are not on stage 0, declare that it is expected with the same.
         if stage == "0" {
-            // Cargo doesn't pass RUSTFLAGS to proc_macros:
-            // https://github.com/rust-lang/cargo/issues/4423
-            // Set `--cfg=bootstrap` explicitly instead.
             cmd.arg("--cfg=bootstrap");
+        } else {
+            cmd.arg("-Zunstable-options");
+            cmd.arg("--check-cfg=values(bootstrap)");
```

Comment:
```suggestion
        }
        cmd.arg("-Zunstable-options");
        cmd.arg("--check-cfg=values(bootstrap)");
```
`--check-cfg` arguments should be always set but because this is a hack I'm not sure.

Reply:
I've done it originally because they [are not needed](https://rust-lang.github.io/rfcs/3013-conditional-compilation-checking.html#enabled-condition-names-are-implicitly-valid), but I guess you'd still lose the scanning for wrong values on stage 0, so I'll change it. It's not *that* much of an issue if the inevitable warning happens on stage1 instead, but better it's early on int the process.